docs/csharp/tutorials/upgrade-to-nullable-references.md

```diff
@@ -0,0 +1,153 @@
+
+```csharp
+public class NewsStoryViewModel
+{
+    public DateTimeOffset Published { get; set; }
+    public string Title { get; set; } = default!;
+    public string Uri { get; set; } = default!;
+}
+```
+
+The assignment of `Title` and `Uri` to `default` which is `null` for the `string` type doesn't change the runtime behavior of the program. The `NewsStoryViewModel` is still constructed with null values, but now the compiler reports no warnings. The **null forgiveness operator**, the `!` character following the `default` expression tells the compiler that the preceding expression is not null. This technique may be expedient when other changes force much larger changes to a code base, but in this application there is a relatively quick and better solution: Make the `NewsStoryViewModel` an immutable type where all the properties are set in the constructor. Make the following changes to the `NewsStoryViewModel`:
+
+[!code-csharp[FinishedViewModel](../../../samples/csharp/tutorials/nullable-reference-migration/finished/SimpleFeedReader/ViewModels/NewsStoryViewModel.cs#FinishedViewModel)]
+
+Once that's done, you need to update the code that configures the automapper so that it uses the constructor rather than setting properties. Open `NewsService.cs` and look for the following code at the bottom of the file:
+
+[!code-csharp[StarterAutoMapper](../../../samples/csharp/tutorials/nullable-reference-migration/start/SimpleFeedReader/Services/NewsService.cs#ConfigureAutoMapper)]
+
+That code maps properties of the `ISyndicationItem` object to the `NewsStoryViewModel` properties. You want the AutoMapper to provide the mapping using a constructor instead. Replace the above code with the following automapper configuration:
+
+[!code-csharp[FinishedViewModel](../../../samples/csharp/tutorials/nullable-reference-migration/finished/SimpleFeedReader/Services/NewsService.cs#ConfigureAutoMapper)]
+
+Notice that because this class is small, and you've examined carefully, you should turn on the `#nullable enable` directive above this class declaration. This change could have broken something, so it's worthwhile to run all the tests and test the application before moving on.
+
+The first set of changes showed you how to discover when the original design indicated that variables should not be set to null. The technique is referred to as **correct by construction**. You guarantee that an object and its properties cannot be null when it is constructed. The compiler's flow analysis provides assurance that those properties are not set to null after construction.
```

Comment:
We're not really guaranteeing anything here though, because there's no validation. AutoMapper could pass in null values.
I'd really like to see that mentioned, in terms of two somewhat different versions of "safe" - compile-time "pretty close to guaranteed, if everything is turned on" vs execution-time verified.